test/mux.js

```diff
@@ -7,177 +7,33 @@ var through = require('through2')
 var debug = require('debug')('multifeed/protodump')
 
 test('Key exchange API', function(t){
-  t.plan(11)
+  t.plan(6)
   var encryptionKey = Buffer.from('deadbeefdeadbeefdeadbeefdeadbeef') // used to encrypt the connection
 
   var mux1 = multiplexer(encryptionKey)
   var mux2 = multiplexer(encryptionKey)
 
   mux1.ready(function(client){
-    mux1.haveFeeds(['foo', 'oof', '03', '01'])
-  })
-
-  mux2.ready(function(client){
-    mux2.haveFeeds(['bar', '02', '01'], {
-      signatures: ['sig'],
-      custom: 'option'
+    mux2.on('manifest', function(m, req) {
+      t.ok(m.keys instanceof Array, 'Manifest contains an array of feed keys')
+      t.deepEqual(m.keys, ['A', 'B', 'C'])
+      req(['A','C','X'])
     })
-  })
-
-  var expectedKeys = ['01', '02', '03', 'foo', 'oof']
-
-  mux1.on('manifest', function(m) {
-    t.ok(m.keys instanceof Array, 'Manifest contains an array of feed keys')
-    t.equal(m.keys[0], 'bar')
-    t.ok(m.signatures instanceof Array, 'Manifest contains a hash of signatures')
-    t.equal(m.signatures[0], 'sig')
-    t.equal(m.custom, 'option')
-
+    var countEv = 0
 
+    // replicate event init missing:
     mux1.on('replicate', function(keys, repl) {
-      // Keys should be alphabetically sorted
-      // and identical on both ends.
-      t.deepEqual(keys, expectedKeys, 'Mux1 replicating same keys')
+      t.deepEqual(keys, ['A','C'], 'List of filtered keys to initialize')
       t.equal(typeof repl, 'function')
+      if (++countEv == 2) t.end()
     })
-
-    mux1.wantFeeds('02', 'oof') // pick some of the remote's keys excluding 'bar'
-  })
-
-  mux2.on('manifest', function(m) {
-    t.equal(m.keys[0], 'foo')
-    t.equal(m.keys[1], 'oof')
     mux2.on('replicate', function(keys, repl) {
-      t.deepEqual(keys, expectedKeys, 'Mux2 replicating same keys')
+      t.deepEqual(keys, ['A','C'], 'List of filtered keys to initialize')
       t.equal(typeof repl, 'function')
+      if (++countEv == 2) t.end()
     })
-
-    mux2.wantFeeds(m.keys) // mark all remote keys as 'want' for classical multicore behaviour
+    mux1.offerFeeds(['A', 'B', 'C'])
   })
-
-
-  pump(
-    mux1.stream,
-    through(function(chunk, _, next) {
-      debug("MUX1->MUX2", chunk.toString('utf8'))
-      this.push(chunk)
-      next()
-    }),
-    mux2.stream,
-    through(function(chunk, _, next) {
-      debug("MUX2->MUX1", chunk.toString('utf8'))
-      this.push(chunk)
-      next()
-    }),
-    mux1.stream
-  )
-
-})
-
-test('Actual replication', function(t) {
```

Comment:
Gone?

Reply:
(rewriting comment cause i accidentally started some kind of github review)

Yes. that test as a bit mislabeled, it was a unit-test that tested the old synchronous have/want handshake which does no longer exist. (It was easy to mock because it didn't do much)

I considered the new async-handshake to be covered by the other tests, writing a separate unit test for it does feel a bit redundant. Thoughts?